sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/ColumnPruningSuite.scala

```diff
@@ -125,5 +125,31 @@ class ColumnPruningSuite extends PlanTest {
     comparePlans(optimized, expected)
   }
 
+  test("Keep broadcast hint when pruning on Join") {
```

Comment:
OK I think it'd make more sense to have this somewhere in some join optimization suite.